bitcoin/src/psbt/mod.rs

```diff
@@ -121,8 +179,102 @@ impl Psbt {
+        self.unsigned_tx.output.iter().map(|o| o.value.to_sat()).sum::<u64>()
+    }
+
+    #[inline]
+    fn get_fee(&self) -> GetFeeResult {
+        let input_value = self.get_input_total_value();
+        let output_value = self.get_output_total_value();
+        match input_value.map(|i| i.checked_sub(output_value)) {
+            Some(Some(fee)) => GetFeeResult::Ok { fee, output_value },
+            Some(None) => GetFeeResult::UnderFlow,
+            None => GetFeeResult::MissingInputValue,
+        }
+    }
+
+    /// The default `max_fee_rate` value used for extracting transactions with [`extract_tx`]
+    ///
+    /// As of 2023, even the biggest overpayers during the highest fee markets only paid around
+    /// 1000 sats/vByte. 25k sats/vByte is obviously a mistake at this point.
+    ///
+    /// [`extract_tx`]: Psbt::extract_tx
+    pub const DEFAULT_MAX_FEE_RATE: f64 = 25000.0_f64;
+
+    /// The default `max_fee_ratio` value used for extracting transactions with [`extract_tx`]
+    ///
+    /// On average, transactions tend to have a 0.3 ratio of fees to output value.
+    /// Looking at transactions around block 807060, it seems that a large majority of fees
+    /// fall under 1.0 (99.99%) Occasionally there will be outliers (especially with small amounts)
+    /// Fee ratio of 10 means "I paid 10x of my send amount + change amount combined in fees."
+    ///
+    /// [`extract_tx`]: Psbt::extract_tx
+    pub const DEFAULT_MAX_FEE_RATIO: f64 = 10.0_f64;
+
+    /// Extracts the [`Transaction`] from a [`Psbt`] by filling in the available signature information.
+    ///
+    /// ## Errors
+    ///
+    /// [`ExtractTxError`] variants will contain either the [`Psbt`] itself or the [`Transaction`]
+    /// that was extracted. These can be extracted from the Errors in order to recover.
+    /// See the error documentation for info on the variants. In general, it covers large fees.
+    pub fn extract_tx(self) -> Result<Transaction, ExtractTxError> {
+        self.internal_extract_tx_with_checks(
+            Self::DEFAULT_MAX_FEE_RATE,
+            Self::DEFAULT_MAX_FEE_RATIO,
+        )
+    }
+
+    /// Extracts the [`Transaction`] from a [`Psbt`] by filling in the available signature information.
+    ///
+    /// ## Errors
+    ///
+    /// See [`extract_tx`].
+    ///
+    /// [`extract_tx`]: Psbt::extract_tx
+    pub fn extract_tx_with_maximum_feerate_and_feeratio(
+        self,
+        max_fee_rate: f64,
+        max_fee_ratio: f64,
+    ) -> Result<Transaction, ExtractTxError> {
+        self.internal_extract_tx_with_checks(max_fee_rate, max_fee_ratio)
+    }
+
+    /// Extracts the [`Transaction`] from a [`Psbt`] by filling in the available signature information
+    /// without performing any checks.
+    ///
+    /// This might pay an absurdly high fee, or be an invalid transaction.
+    ///
+    /// ## Safety
+    ///
+    /// This function should only be called when the fee of the transaction is irrelevant.
```

Comment:
This is not a valid use of the `unsafe` keyword. Unsafety in Rust refers specifically to memory safety.